### DISCUSSION TITLE
Set day ranges with click-move-release

### DISCUSSION THREAD

**Discussion by Eric24** [gpbl/react-day-picker#1662] [Ideas]
**Is your feature request related to a problem? Please describe.**
It is more common these days to see calendar ranges selected by clicking on the first day, holding the mouse button down, moving to the second day, and releasing. The current method in react-day-picker of separately clicking the start and end days is not what people are expecting (based on feedback I've received).

**Describe the solution you'd like**
I think from a UI/UX perspective, both methods could existing simultaneously, since the click-move-release gesture only applies for a range select operation. So I propose that this is "always on", so that the user could use either method (two separate clicks or one click-move-release).

**Describe alternatives you've considered**
Currently, we just use the two-separate-clicks method. I don't think this is something can be implemented externally to react-day-picker.

**Additional context**
1. Note that I don't think this is a "drag and drop" operation. Nothing is being dragged. It's simply a matter of allowing the cursor to be moved over the calendar while the button is pressed, and recalculating the date range dynamically based on the earliest and latest day that have been "hit" during the mouse-down state.
2. The same functionality should apply equally to a touch interface, where tapping two days is the currently supported method and touching one day and then moving to the second day is this proposed method.

**Comment by gpbl** [comment]
@Eric24 this is a great feedback, thanks! 

We are not going to add new features to v7 – as I'm all working on the next major release. 

>  I don't think this is something can be implemented externally to react-day-picker.

It would be great to see if (with the current v8 API) it is already possible.  I put this in the v8 milestones: if you have some times to check it out it would be great!

**Comment by gpbl** [comment]
I put this in documentation for v8.1.0 – but it should be relatively easy to implement using a Day custom component:

https://react-day-picker-next.netlify.app/docs/custom-components

It would be nice to see actually _how easy_ it is to implement that!

**Comment by wiesson** [comment]
> It is more common these days to see calendar ranges selected by clicking on the first day, holding the mouse button down, moving to the second day, and releasing.

I like this feature but I have to admit that I have never seen it so far - do you have an example?
